common-setup.sh

```diff
@@ -141,3 +141,39 @@ try_reflex_shell() {
     prebuild_try_reflex_shell
     nix-shell "$DIR/gc-roots/shell.drv" $NIXOPTS "$@"
 }
+
+enable_cache() {
+    nixconf_dir="/etc/nix"
+    nixconf="$nixconf_dir/nix.conf"
+    our_cache="https://nixcache.reflex-frp.org"
+    caches_line="binary-caches = https://cache.nixos.org $our_cache"
+    our_key="JJiAKaRv9mWgpVAz8dwewnZe0AzzEAzPkagE9SP5NWI="
+    keys_line="binary-cache-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= ryantrinkle.com-1:$our_key"
+    sudo_msg="This requires root access."
+    if [ ! -e "$nixconf" ] || ! grep -q 'binary-caches\|binary-cache-public-keys\|binary-caches-parallel-connections' "$nixconf" ; then
+	if [ ! -e "$nixconf" ];
+	then echo "Creating $nixconf - $sudo_msg";
+	else echo "Adding cache settings to $nixconf - $sudo_msg";
+	fi;
+	sudo mkdir -p "$nixconf_dir"
+	sudo tee -a "$nixconf" > /dev/null <<EOF
+$caches_line
+$keys_line
+binary-caches-parallel-connections = 40
+EOF
+    elif ! grep -q "$our_cache" "$nixconf" || ! grep -q "$our_key" "$nixconf" ; then
+	backup="$nixconf.$(date -u +"%FT%TZ").bak"
+	echo "$nixconf already exists: creating backup - $sudo_msg"
+	sudo cp "$nixconf" "$backup"
+	echo "backup saved at $backup"
+	echo "adding caches"
+	if ! grep -q "$our_cache" "$nixconf" ; then
+            sudo sed -i.bak 's|^\(binary-caches[ =].*\)$|\1 '"$our_cache"'|' "$nixconf"
+	fi
+	if ! grep -q "$our_key" "$nixconf" ; then
+            sudo sed -i.bak 's|^\(binary-cache-public-keys[ =].*\)$|\1 ryantrinkle.com-1:'"$our_key"'|' "$nixconf"
+	fi
+    fi
+}
+
+enable_cache
```

Comment:
Is this safe to do in all OS or should it only run when required (e.g. Mac OS)?

Reply:
/etc/nix/nix.conf should never be edited directly on NixOS; however, we can also assume that users on NixOS are slightly more savvy with Nix, so it should be sufficient to ask them to do something like [this](https://github.com/reflex-frp/reflex-platform/blob/develop/notes/NixOS.md) to enable caches.

Reply:
I'm not sure enable_cache should go here.  It should go along with the stuff that detects that nix isn't installed and then installs it if need be (though this should be run even if nix is already installed).